Code/Framework/AzCore/AzCore/Metrics/JsonTraceEventLogger.cpp

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/IO/ByteContainerStream.h>
+#include <AzCore/IO/TextStreamWriters.h>
+#include <AzCore/IO/GenericStreams.h>
+#include <AzCore/JSON/writer.h>
+#include <AzCore/JSON/RapidjsonAllocatorAdapter.h>
+#include <AzCore/Metrics/JsonTraceEventLogger.h>
+#include <AzCore/std/parallel/scoped_lock.h>
+#include <AzCore/std/ranges/repeat_view.h>
+
+namespace AZ::Metrics
+{
+    constexpr AZStd::string_view NameKey = "name";
+    constexpr AZStd::string_view CategoryKey = "cat";
+    constexpr AZStd::string_view PhaseKey = "ph";
+    constexpr AZStd::string_view IdKey = "id";
+    constexpr AZStd::string_view TimestampKey = "ts";
+    constexpr AZStd::string_view ThreadTimestampKey = "tts";
+    constexpr AZStd::string_view ProcessIdKey = "pid";
+    constexpr AZStd::string_view ThreadIdKey = "tid";
+
+    // Format strings for the TraceEvent Format
+    constexpr AZStd::string_view ArrayStart = "[";
+    constexpr AZStd::string_view ArrayEnd = "]";
+    constexpr AZStd::string_view ObjectStart = "{";
+    constexpr AZStd::string_view ObjectEnd = "}";
+    constexpr AZStd::string_view Newline = "\n";
+
+    constexpr int32_t IndentStep = 2;
+    constexpr size_t StackAllocatorSize = 2048;
+
+    using EventJsonWriter = rapidjson::Writer<AZ::IO::RapidJSONWriteStreamNoCache, rapidjson::UTF8<char>, rapidjson::UTF8<char>,
+        AZ::Json::RapidjsonStackAllocator<StackAllocatorSize>>;
+
+    static void AppendNewlineWithIndent(JsonTraceEventLogger::JsonEventString& eventString, int32_t indent, bool prependComma)
+    {
+        if (prependComma)
+        {
+            eventString += ',';
+        }
```

Comment:
nit:
You can save an extra '+=' operation by defining an additional 
```
const AZStd::string_view CommaNewLine = ",\n";
```
and change this to
```
eventString += prependComma?CommaNewLine:NewLine;
```

Reply:
Yeah I'll do that